### DISCUSSION TITLE
Wireless Controller (DS4) not working in Archlinux

### DISCUSSION THREAD

**Discussion by rebelinux** [flyinghead/flycast#1685] [General]
**Please Note: This form is the minimum required information for submitting bugs.**  
**Removing this form may lead to your issue being closed until it is completed.**

Platform / OS / Hardware: 

Archlinux \ Dell XPS 13 9380

Github hash: 

Hardware: 

I have a Dual Shock 4 connected via BlueToodth. The controller is detected as Wireless Controller but I can't use it in the game. When I try to configure the mapping it doesn't work either. If I connect the controller via USB it works. The strange thing is that the Bluetoodth control worked at some point.

**_Debugging Steps Tested_**

  * Connect the controller via Bluetoodth

![image](https://github.com/user-attachments/assets/86673d28-63a3-4300-85e4-d5cdaaa121da)

**_Logs Gathered_**

```

How can I collect the logs?


```

**_Screenshots_**

![image](https://github.com/user-attachments/assets/1a9aa34a-0c6e-480f-a541-bb6582fbc20c)

**Comment by rebelinux** [comment]
Dmesg:

```
[  649.625835] usbcore: registered new interface driver usbhid
[  649.625836] usbhid: USB HID core driver
[ 4736.978799] usb 1-1: USB disconnect, device number 5
[ 5107.009562] playstation 0005:054C:09CC.0005: unknown main item tag 0x0
[ 5107.010285] playstation 0005:054C:09CC.0005: hidraw2: BLUETOOTH HID v1.00 Gamepad [Wireless Controller] on 9c:b6:d0:97:d5:8c
[ 5107.036504] input: Wireless Controller as /devices/virtual/misc/uhid/0005:054C:09CC.0005/input/input30
[ 5107.036765] input: Wireless Controller Motion Sensors as /devices/virtual/misc/uhid/0005:054C:09CC.0005/input/input31
[ 5107.036893] input: Wireless Controller Touchpad as /devices/virtual/misc/uhid/0005:054C:09CC.0005/input/input32
[ 5107.037029] playstation 0005:054C:09CC.0005: Registered DualShock4 controller hw_version=0x0000b40c fw_version=0x0000a007
```

**Comment by flyinghead** [comment]
I'm afraid this is an SDL issue. Flycast relies entirely on it to access input devices.

You can run Flycast from the command line and it will print out its log to the console.

**Comment by cinnamonmatexfce** [comment]
@rebelinux something's going on with ArchLinux itself (or SDL) I presume...

Also I am in trouble with DualShock4 via bluetooth, but on Steam: the gamepad is recognized by Steam itself, but all tested games are NOT recognizing it.

Did you try downgrading `sdl2` to *2.30.7‐1* or *2.30.6‐1* ?

Until the weekend I won't have a chance to try it on my installation unfortunately

**Comment by rebelinux** [comment]
> @rebelinux something's going on with ArchLinux itself (or SDL) I presume...
> 
> Also I am in trouble with DualShock4 via bluetooth, but on Steam: the gamepad is recognized by Steam itself, but all tested games are NOT recognizing it.
> 
> Did you try downgrading `sdl2` to _2.30.7‐1_ or _2.30.6‐1_ ?
> 
> Until the weekend I won't have a chance to try it on my installation unfortunately

I have downgraded the sdl package but I don't know if the flackpak uses the locally installed sdl libraries or a different package when the flackpak was compiled.

**Comment by rebelinux** [comment]
i get it working again. it seem a bluez bug

https://github.com/bluez/bluez/issues/372
https://github.com/blueman-project/blueman/issues/2255

i can reproduce the behaivior and get controller connected

**Comment by rebelinux** [comment]
I do further debuging and found the real issue 

The extension gnome-bluetooth-quick-connect cause some issue with the gamepad when i uninstalled the extension the gamepad work as expected. :)

https://github.com/Extensions-Valhalla/gnome-bluetooth-quick-connect
